src/Rector/MethodCall/WhereToWhereLikeRector.php

```diff
@@ -0,0 +1,180 @@
+        'where' => 'whereLike',
+        'orwhere' => 'orWhereLike',
+    ];
+
+    private bool $usingPostgresDriver = false;
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Changes `where` method calls to `whereLike` method calls in the Eloquent & Query Builder', [
+                new ConfiguredCodeSample(
+                    <<<'CODE_SAMPLE'
+$query->where('name', 'like', 'Rector');
+$query->orWhere('name', 'like', 'Rector');
+$query->where('name', 'like binary', 'Rector');
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+$query->whereLike('name', 'Rector');
+$query->orWhereLike('name', 'Rector');
+$query->whereLike('name', 'Rector', true);
+CODE_SAMPLE
+                    ,
+                    ['usingPostgresDriver' => false]
+                ),
+                new ConfiguredCodeSample(
+                    <<<'CODE_SAMPLE'
+$query->where('name', 'ilike', 'Rector');
+$query->orWhere('name', 'ilike', 'Rector');
+$query->where('name', 'like', 'Rector');
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+$query->whereLike('name', 'Rector');
+$query->orWhereLike('name', 'Rector');
+$query->whereLike('name', 'Rector', true);
+CODE_SAMPLE
+                    ,
+                    ['usingPostgresDriver' => true]
+                ),
+            ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param  MethodCall  $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->isObjectType($node->var, new ObjectType('Illuminate\Database\Query\Builder'))) {
+            return null;
+        }
+
+        if (! in_array($this->getLowercaseMethodCallName($node), array_keys(self::WHERE_LIKE_METHODS), true)) {
+            return null;
+        }
+
+        if (count($node->getArgs()) !== 3) {
+            return null;
+        }
+
+        $likeParameter = $this->getLikeParameterUsedInQuery($node);
+
+        if (! in_array($likeParameter, ['like', 'like binary', 'ilike', 'not like', 'not like binary', 'not ilike'], true)) {
+            return null;
+        }
+
+        $this->setNewNodeName($node, $likeParameter);
+
+        $this->setCaseSensitivity($node, $likeParameter);
+
+        // Remove the second argument (the 'like' operator)
+        unset($node->args[1]);
+
+        return $node;
+    }
+
+    public function configure(array $configuration): void
+    {
+        if ($configuration === []) {
+            $this->usingPostgresDriver = false;
+
+            return;
+        }
+
+        Assert::count($configuration, 1);
+        Assert::keyExists($configuration, 'usingPostgresDriver');
+        Assert::boolean($configuration['usingPostgresDriver']);
+        $this->usingPostgresDriver = $configuration['usingPostgresDriver'];
```

Comment:
It might be useful to introduce a const for the array key that's public or just have a static method that creates the array e.g. `WhereToWhereLikeRector::usingPostgresDriver()` Then update the code sample for it.